### PR TITLE
Octopus board upstream

### PR DESCRIPTION
**Pull request by FPlohl**
Add support for Norik Octopus SoM & IO-Board (nRF9160).

### CODE REVIEW

File: boards/norik/octopus_som/doc/img/Norik_Octopus_Som_front.png
Comment (kartben) [inline boards/norik/octopus_som/doc/img/Norik_Octopus_Som_front.png:1]: ditto re: webp and trying to have a transparent bg

File: boards/norik/octopus_som/norik_octopus_som_common.dtsi [hunk 1]
```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2023-2024 Norik Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "norik_octopus_som_common-pinctrl.dtsi"
+
+/ {
+   	model = "Norik Octopus SoM";
+   	compatible = "norik,octopus-som";
```
Comment (kartben) [inline boards/norik/octopus_som/norik_octopus_som_common.dtsi:10]: please add norik to vendor-prefixes.txt file

**Review by kartben — CHANGES_REQUESTED** [review]

**Review by kanakshilledar — CHANGES_REQUESTED** [review]
Please signoff your commits in the following format
`Signed-off-by: Your Name <your.email@example.com>`

There are a lot of formatting errors in your files. Please get rid of all the trailing whitespace in most of your files. Also add an empty newline at the end of files. Also please maintain consistency in TABS and SPACES in the devicetree.
You can refer this [Compliance Check action](https://github.com/zephyrproject-rtos/zephyr/actions/runs/8481513627/job/23239119759?pr=70902) to locate exactly where you need to do the changes.

File: boards/norik/octopus_som/Kconfig.defconfig [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# must be linked into the code-partition (non-secure) defined in DT, regardless.
+# Apply this configuration below by setting the Kconfig symbols used by
+# the linker according to the information extracted from DT partitions.
+
+# Workaround for not being able to have commas in macro arguments
+DT_CHOSEN_Z_CODE_PARTITION := zephyr,code-partition
+
+config FLASH_LOAD_SIZE
+	default $(dt_chosen_reg_size_hex,$(DT_CHOSEN_Z_CODE_PARTITION))
+	depends on BOARD_NORIK_OCTOPUS_SOM && TRUSTED_EXECUTION_SECURE
```
Comment (nordicjm) [inline boards/norik/octopus_som/Kconfig.defconfig:21]: Kconfig here doesn't look right

File: boards/norik/octopus_som/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+Programming and Debugging
+*************************
+
+Norik Octopus SoM can be programmed and debugged using the exposed SWD pins.
+
+Building an application
+=======================
+
+In most case you'll need to use ``ns`` target for building examples. 
+Some examples don't require non secure mode and can be built without ``ns`` target.
```
Comment (nordicjm) [inline boards/norik/octopus_som/doc/index.rst:116]: use full board target names

File: boards/norik/octopus_som/norik_octopus_som_common-pinctrl.dtsi [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+			psels = <NRF_PSEL(TWIM_SDA, 0, 4)>,
+					<NRF_PSEL(TWIM_SCL, 0, 3)>;
+		};
+	};
+
+	i2c0_sleep: i2c0_sleep {
+		group1 {
+			psels = <NRF_PSEL(TWIM_SDA, 0, 4)>,
+					<NRF_PSEL(TWIM_SCL, 0, 3)>;
+			low-power-enable;
```
Comment (nordicjm) [inline boards/norik/octopus_som/norik_octopus_som_common-pinctrl.dtsi:32]: indents for all of these look off

File: boards/norik/octopus_som/norik_octopus_som_common.dtsi [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+	   zephyr,shell-uart = &uart0;
+	   zephyr,uart-mcumgr = &uart0;
+   	};
+
+   	leds {
+	   compatible = "gpio-leds";
+	   led0: led_0 {
+		   gpios = <&gpio0 7 0>;
+		   label = "Green LED 1";
+	   };
```
Comment (nordicjm) [inline boards/norik/octopus_som/norik_octopus_som_common.dtsi:23]: tab indents

File: boards/norik/octopus_som/norik_octopus_som_common.dtsi [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+	   led0 = &led0;
+	   pwm-led0 = &pwm_led0;
+	   watchdog0 = &wdt0;
+	   simselect = &sim_select;
+   };
+};
+
+&adc {
+   status = "okay";
+};
```
Comment (nordicjm) [inline boards/norik/octopus_som/norik_octopus_som_common.dtsi:53]: etc.

File: boards/norik/octopus_som/norik_octopus_som_common.dtsi [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+	cs-gpios = <&gpio0 11 GPIO_ACTIVE_LOW>;
+
+	pinctrl-0 = <&spi3_default>;
+	pinctrl-1 = <&spi3_sleep>;
+	pinctrl-names = "default", "sleep";
+	mc3635: mc3635@0 {
+		compatible = "vnd,spi-device";
+		spi-max-frequency = <1000000>;
+		reg = <0>;
+//		int1-gpios = <&gpio0 12 0>;
```
Comment (nordicjm) [inline boards/norik/octopus_som/norik_octopus_som_common.dtsi:83]: why is this commented?

File: boards/norik/octopus_io_board/Kconfig.defconfig [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+# must be linked into the code-partition (non-secure) defined in DT, regardless.
+# Apply this configuration below by setting the Kconfig symbols used by
+# the linker according to the information extracted from DT partitions.
+
+# Workaround for not being able to have commas in macro arguments
+DT_CHOSEN_Z_CODE_PARTITION := zephyr,code-partition
+
+config FLASH_LOAD_SIZE
+	default $(dt_chosen_reg_size_hex,$(DT_CHOSEN_Z_CODE_PARTITION))
+	depends on BOARD_NORIK_OCTOPUS_IO_BOARD && TRUSTED_EXECUTION_SECURE
```
Comment (nordicjm) [inline boards/norik/octopus_io_board/Kconfig.defconfig:21]: previous commit comments apply to this commit also

**Review by nordicjm — CHANGES_REQUESTED** [review]

**Comment by kartben** [comment]
@FPlohl are you still interested in moving this PR forward? Please let us know if you have any questions

**Comment by kartben** [comment]
@FPlohl any news on this?

**Comment by FPlohl** [comment]
We had to postpone PR due to other projects. Now we're working on it again.

**Pull request closed without merging**